posthog/src/main/java/com/posthog/internal/replay/RRUtils.kt

```diff
@@ -4,11 +4,18 @@ import com.posthog.PostHog
 import com.posthog.PostHogInternal
 
 @PostHogInternal
-public fun List<RREvent>.capture() {
+public enum class PostHogSessionReplayMode(public val mode: String) {
+    WIREFRAME("wireframe"),
+    SCREENSHOT("screenshot"),
+}
+
+@PostHogInternal
+public fun List<RREvent>.capture(mode: PostHogSessionReplayMode = PostHogSessionReplayMode.WIREFRAME) {
     val properties =
         mutableMapOf(
             "\$snapshot_data" to this,
             "\$snapshot_source" to "mobile",
+            "\$snapshot_mode" to mode.mode,
```

Comment:
maybe `source` and `mode` should only be part of the full snapshot?

technically the `mode` can also be inferred server side since we only send `wireframe.type=screenshot`.

the SDK and platform can also be inferred from `$lib`, so there is no need for something else here, eg `posthog-android`, `posthog-ios`, `posthog-react-native`, etc.

Reply:
> maybe source and mode should only be part of the full snapshot?

if we do the same as snapshot source then we'll take the first not-null value we receive in a session as the only value for that property. so safest to be on everything for debugging, but safe to be on only some messages

(with the assumption that very few or no sessions will span web and mobile or wireframe and snapshot)

> technically the mode can also be inferred server side 

ideally we'd keep making ingestion less dependent on the content of the messages - i.e. we can't extra compress the web messages right now because we need to read the contents in blobby. so ideally we'd be explicit here - doesn't help old versions obvs but better in the long run IMHO

Reply:
What's about the naming `$snapshot_mode` and as a sibling of `$snapshot_source`, is that ok?
Are we ok to infer the SDK from `$lib` as well or should we be more explicit here?

Reply:
yep mode is good...
so long as we're sending $lib then i think we need that to be unique anyway so we're good there